src/libraries/System.IO.Compression/tests/CompressionStreamUnitTests.ZLib.cs

```diff
@@ -16,5 +18,44 @@ public class ZLibStreamUnitTests : CompressionStreamUnitTestBase
         public override Stream CreateStream(Stream stream, CompressionLevel level, bool leaveOpen) => new ZLibStream(stream, level, leaveOpen);
         public override Stream BaseStream(Stream stream) => ((ZLibStream)stream).BaseStream;
         protected override string CompressedTestFile(string uncompressedPath) => Path.Combine("ZLibTestData", Path.GetFileName(uncompressedPath) + ".z");
+
+        [Fact]
+        public void StreamCorruption_IsDetected()
```

Comment:
Shouldn't we add this test to `CompressionStreamUnitTests.Deflate.cs` as well?

Reply:
Sadly this is not really an option as `DeflateStream` does not do CRC, hence fails to detect stream corruption (unlike zlib and gzip streams). Truncation detection works fine though.